### PR TITLE
Add outline and size property to ButtonLink, improve CSS structure

### LINKED ISSUE #29: Improve <ButtonLink /> Component
**What's the problem you want solved?**
Currently you need to pass class names to get, for example, the outlined version of the ButtonLink. This is not handy and too hard to memorise.

Example:
```
<ButtonLink to="/foerderungen" className="btn-brand-outline mt-5">
  Zur Förderübersicht
</ButtonLink>
```

**Is there a solution you'd like to recommend?**

New version should be:
```
<ButtonLink to="/foerderungen" outline className="mt-5">
  Zur Förderübersicht
</ButtonLink>
```

### PR DESCRIPTION
**Pull request by ohrie**
## Proposed changes

Add properties to <ButtonLink />` for outline option and size.
Also restructure CSS classes for better maintaining.

## Closes Issue
Closes #29

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/Links/ButtonLink.tsx [hunk 1]
```diff
@@ -2,16 +2,32 @@ import classNames from 'classnames';
...
+  outline?: boolean;
+};
 
 // TODO: Refactor all those link components to share more logic
 export const ButtonLink: React.FC<Props> = ({
   to,
   className = '',
   newWindow = false,
+  outline = false,
+  size = 'md',
```
Comment (tordans) [inline src/components/Links/ButtonLink.tsx:21]: I feel like true/false props would work better here. `<ButtonLink big outline to…>`.

You could create Props that make sure that only one (big or small or none) is selected. However, that's a bit much IMO.

File: src/components/Links/ButtonLink.tsx [hunk 2]
```diff
@@ -2,16 +2,32 @@ import classNames from 'classnames';
...
   children,
 }) => {
-  const styles = classNames(className, 'btn-brand');
+  const styles = classNames(
+    className,
+    { 'btn-brand-outline': outline },
+    { 'w-full px-8 py-3 md:text-lg': size === 'lg' },
+    { 'px-3 py-1 text-sm font-medium': size === 'sm' },
+    'btn-brand'
```
Comment (tordans) [inline src/components/Links/ButtonLink.tsx:29]: I think that would be the perfect situation to move the styles of btn-brand back here. We only use it here ([Search](https://github.com/FixMyBerlin/fixmy.rsv/search?q=btn-brand)), so there is no reason to extract it in a "css"-component. The rule of thumb for Tailwind CSS is: Keep it inside the component until you re-use it somewhere or extensively.

File: src/components/Links/ButtonLink.tsx [hunk 3]
```diff
@@ -2,16 +2,32 @@ import classNames from 'classnames';
...
   className = '',
   newWindow = false,
+  outline = false,
+  size = 'md',
   children,
 }) => {
-  const styles = classNames(className, 'btn-brand');
+  const styles = classNames(
+    className,
+    { 'btn-brand-outline': outline },
```
Comment (tordans) [inline src/components/Links/ButtonLink.tsx:26]: See below.

**Review by tordans — COMMENTED** [review]
Looks good. Its the right situation to extract this now, IMO!
Suggestions on the "how" attached :-).

**Comment by tordans** [comment]
Off topic on the branch name "feature/button-modularizing": It that really a feature or a refactoring? … – Avoid spending time to think about this is one of the reasons why I try to avoid those semantics when I can :-).

**Comment by ohrie** [comment]
> Off topic on the branch name "feature/button-modularizing": It that really a feature or a refactoring?

You're right. Didn't think about it properly :)

**Pull request merged**